### DISCUSSION TITLE
How to upload files bigger than 25MB?

### DISCUSSION THREAD

**Discussion by OS-Scratch** [community/community#147296] [Repositories]
### Select Topic Area

General

### Body

Hello, I am developing an OS in TurboWarp. When saving the file (27.9 MB in size) and uploading it, I get an error. I really need to upload it using the browser as I'm not familiar with Terminal and Command Prompt etc. How can I do that?

**Comment by NouarSalhEddine** [comment]
TurboWarp (and many browser-based tools) often have file size limits for uploads. A 27.9 MB file might be too large for the platform to handle directly through the browser. This is pretty common with browser-based tools, as they’re designed to handle smaller files for performance and security reasons.

What You Can Do:
Here are some simple, browser-friendly solutions to get your file uploaded:

1. Compress the File
If your file isn’t already compressed, you can reduce its size by zipping it. This might make it small enough to upload directly.

On Windows:

Right-click the file > Select “Send to” > “Compressed (zipped) folder”.

On macOS:

Right-click the file > Select “Compress”.

Try uploading the zipped file and see if it works.

2. Split the File
If compressing doesn’t reduce the size enough, you can split the file into smaller chunks. This way, you can upload them one by one.

Use a free tool like 7-Zip (Windows) or Keka (macOS) to split the file into smaller parts.

After uploading, you can recombine the parts on the server or wherever you’re storing the file.

3. Use a Cloud Storage Service
If TurboWarp itself has a file size limit, you can upload your file to a cloud storage service like Google Drive, Dropbox, or OneDrive.

Once uploaded, share the file’s link and use that link in TurboWarp (if it supports linking to external files).

4. Check TurboWarp’s Documentation
TurboWarp might have specific guidelines or tools for handling large files. Check their official documentation or forums for tips on uploading larger files.

5. Use a Browser Extension for Large Uploads
Some browser extensions can help with large file uploads by breaking them into smaller chunks. Look for extensions like “Advanced Uploader” or similar tools that work with your browser.

6. Ask for Help from the TurboWarp Community
If none of the above works, reach out to the TurboWarp community or support team. They might have a workaround or can increase the upload limit for your account.

If You’re Open to Trying Something New:
If you’re willing to explore a slightly more technical (but still simple) solution, you could use a free file transfer service like WeTransfer or File.io. These services are designed for large files and are super easy to use:

Go to [WeTransfer](https://wetransfer.com/) or [File.io](https://file.io/).

Upload your file.

Share the download link with yourself or anyone who needs it.

Final Thoughts:
The easiest solution is probably compressing the file or using a cloud storage service. Both are straightforward and don’t require any technical know-how. If you’re still stuck, let me know, and we can brainstorm further! You’ve got this! 😊

**Reply by OS-Scratch** [reply]
I have TB Desktop, which doesn't have limits. And where is the community? Only if you didn't know, TurboWarp is an editor for the Scratch programming language.

**Comment by FirdavsToshniyozov** [comment]
fucking slow application. Impossible to upload 1.9gb file. daamn

**Reply by OS-Scratch** [reply]
What Application are you refering to? GitHub or Scratch / Turbowarp?
